json_encoder_test.go

```diff
@@ -36,16 +36,37 @@ import (
 )
 
 func assertJSON(t *testing.T, expected string, enc *jsonEncoder) {
-	assert.Equal(t, expected, string(enc.bytes), "Encoded JSON didn't match expectations.")
+	encoded := strings.TrimPrefix(string(enc.bytes), "{")
+	assert.Equal(t, expected, encoded, "Encoded JSON didn't match expectations.")
 }
 
 func withJSONEncoder(f func(*jsonEncoder)) {
 	enc := newJSONEncoder()
-	enc.AddString("foo", "bar")
 	f(enc)
 	enc.Free()
 }
 
+type testBuffer struct {
+	*bytes.Buffer
+}
+
+func newTestBuffer() *testBuffer {
```

Comment:
if you're returning a pointer here, you can embed the `bytes.Buffer` without a pointer. the other option is to use a value for `testBuffer` and leave `*bytes.Buffer` as a pointer. I personally prefer the first, since you could just do `&testBuffer{}` and use it without needing `newTestBuffer`